pulp_container/app/access_policy.py

```diff
@@ -43,3 +46,101 @@ def has_namespace_obj_perms(self, request, view, action, permission):
+    """
+    Access policy for ContainerRepositoryVersionViewSet which handles namespace permissions.
+    """
+
+    def has_model_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has a model-level permission.
+
+        Args:
+            request (rest_framework.request.Request): The request being made.
+            view (subclass rest_framework.viewsets.GenericViewSet): The view being checked for
+                authorization.
+            action (str): The action being performed, e.g. "destroy".
+            permission (str): The name of the Permission to be checked. In the form
+                `app_label.codename`, e.g. "core.delete_task".
+
+        Returns:
+            True if the user has the Permission named by the ``permission`` argument at the model-level.
+                False otherwise.
+        """
+        return request.user.has_perm(permission)
+
+    def has_repoversion_repo_obj_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has object-level permission on the ``repository`` object.
+
+        Args:
+            request (rest_framework.request.Request): The request being made.
+            view (subclass rest_framework.viewsets.GenericViewSet): The view being checked for
+                authorization.
+            action (str): The action being performed, e.g. "destroy".
+            permission (str): The name of the Permission to be checked. In the form
+                `app_label.codename`, e.g. "core.delete_task".
+
+        Returns:
+            True if the user has the Permission named by the ``permission`` argument on the
+            ``repository`` parameter at the object-level. False otherwise.
+        """
+        repo = view.get_object().repository
+        return request.user.has_perm(permission, repo)
+
+    def has_repoversion_repo_model_or_obj_perms(self, request, view, action, permission):
+        """
+        Checks if the current user has model-level or object-level permissions on the ``repository``.
+
+        Args:
+            request (rest_framework.request.Request): The request being made.
+            view (subclass rest_framework.viewsets.GenericViewSet): The view being checked for
+                authorization.
+            action (str): The action being performed, e.g. "destroy".
+            permission (str): The name of the Permission to be checked. In the form
+                `app_label.codename`, e.g. "core.delete_task".
+
+        Returns:
+            True if the user has the Permission named by the ``permission`` at the model-level or on the
+                argument on the ``repository`` parameter at the object-level. False otherwise.
+        """
+        if self.has_model_perms(request, view, action, permission):
+            return True
+        if self.has_repoversion_repo_obj_perms(request, view, action, permission):
+            return True
+        return False
```

Comment:
@bmbouter , I wrote those in a generic way, should I add them to the global ones in pulpcore right away? pulp_container requires 3.10 anyway.

Shorter name which describes that it can be used for repo versions only and that repo perms needs to be provided are welcome.

Reply:
Will move to pulpcore